servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ContentCodingTest.java

```diff
@@ -0,0 +1,384 @@
+                {of(none(), deflate()), null, none(), true, Protocol.H2},
+                {of(gzip()), of(none()), none(), true, Protocol.H1},
+                {of(gzip()), of(none()), none(), true, Protocol.H2},
+                {of(gzip()), of(gzip(), none()), none(), true, Protocol.H1},
+                {of(gzip()), of(gzip(), none()), none(), true, Protocol.H2},
+                {of(gzip()), of(gzip(), none()), none(), true, Protocol.H1},
+                {of(gzip()), of(gzip(), none()), none(), true, Protocol.H2},
+                {of(gzip()), of(gzip(), none()), gzip(), true, Protocol.H1},
+                {of(gzip()), of(gzip(), none()), gzip(), true, Protocol.H2},
+                {null, of(gzip(), none()), gzip(), false, Protocol.H1},
+                {null, of(gzip(), none()), gzip(), false, Protocol.H2},
+                {null, of(gzip(), deflate(), none()), deflate(), false, Protocol.H1},
+                {null, of(gzip(), deflate(), none()), deflate(), false, Protocol.H2},
+                {null, of(gzip(), none()), none(), true, Protocol.H1},
+                {null, of(gzip(), none()), none(), true, Protocol.H2},
+        };
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ASYNC_ERROR.set(false);
+        try {
+            client.close();
+        } finally {
+            serverContext.close();
+        }
+    }
+
+    private ServerContext listenAndAwait() throws Exception {
+        HttpProtocolConfig config = testEncodingScenario.protocol.build(testEncodingScenario.serverSupported);
+
+        StreamingHttpService service = (ctx, request, responseFactory) -> Single.succeeded(responseFactory.ok()
+                .payloadBody(from(payload((byte) 'b')), textSerializer()));
+
+        StreamingHttpServiceFilterFactory filterFactory = REQ_RESP_VERIFIER.apply(testEncodingScenario);
+        return httpServerBuilder.appendServiceFilter(filterFactory)
+                .protocols(config)
+                .listenStreamingAndAwait(service);
+    }
+
+    private HttpClient newClient() {
+        HttpProtocolConfig config = testEncodingScenario.protocol.build(testEncodingScenario.clientSupported);
+
+        return HttpClients.forSingleAddress(serverHostAndPort(serverContext))
+                .protocols(config)
+                .build();
+    }
+
+    @Test
+    public void test() throws Exception {
+        if (expectedSuccess) {
+            assertSuccessful(testEncodingScenario.requestEncoding);
+        } else {
+            assertNotSupported(testEncodingScenario.requestEncoding);
+        }
+    }
+
+    private static String payload(byte b) {
+        byte[] payload = new byte[PAYLOAD_SIZE];
+        Arrays.fill(payload, b);
+        return new String(payload, StandardCharsets.US_ASCII);
+    }
+
+    private void assertSuccessful(final ContentCoding encoding) throws Exception {
+        assertResponse(client.request(client
+                .get("/")
+                .encoding(encoding)
```

Comment:
Can client-side users configure compression on the builder and then have it for all requests?

Reply:
They can't as of now. My thinking was that a request isn't always a good fit for compression, eg. GET/DELETE requests, or even smaller PUTs/POSTs. It looked more fluent to enable it on-demand, rather than disable it on-demand. If you have hard objections about it, I can track it as a separate PR.

Reply:
Discussed offline, users can insert a simple filter if they need compression for all requests that are sent.